render-test/runner.cpp

```diff
@@ -814,9 +896,77 @@ bool TestRunner::runOperations(const std::string& key, TestMetadata& metadata) {
             return false;
         }
         frontend.getRenderer()->removeFeatureState(sourceID, sourceLayer, featureID, stateKey);
+    } else if (operationArray[0].GetString() == panGestureOp) {
+        // benchmarkPanGesture
+        assert(operationArray.Size() >= 4u);
+        assert(operationArray[1].IsString()); // identifier
+        assert(operationArray[2].IsNumber()); // duration
+        assert(operationArray[3].IsArray());  // start [lat, lng, zoom]
+        assert(operationArray[4].IsArray());  // end [lat, lng, zoom]
+
+        if (metadata.mapMode != mbgl::MapMode::Continuous) {
+            metadata.errorMessage = "Map mode must be Continous for " + panGestureOp + " operation";
+            return false;
+        }
+
+        std::string mark = operationArray[1].GetString();
+        int duration = operationArray[2].GetFloat();
+        LatLng startPos, endPos;
+        double startZoom, endZoom;
+        std::vector<double> samples;
+
+        auto parsePosition = [](auto arr) -> std::tuple<LatLng, double> {
+            assert(arr.Size() >= 3);
+            return {{arr[1].GetDouble(), arr[0].GetDouble()}, arr[2].GetDouble()};
+        };
+
+        std::tie(startPos, startZoom) = parsePosition(operationArray[3].GetArray());
+        std::tie(endPos, endZoom) = parsePosition(operationArray[4].GetArray());
+
+        // Jump to the starting point of the segment and make sure there's something to render
+        map.jumpTo(mbgl::CameraOptions().withCenter(startPos).withZoom(startZoom));
+
+        MapBecomeIdleObserver observer;
+
+        while (!observer.finishRenderingMap) {
+            frontend.renderOnce(map);
+        }
+
+        if (observer.mapLoadFailure) return false;
+
+        size_t frames = 0;
+        double totalTime = 0.0;
+        bool transitionFinished = false;
+
+        mbgl::AnimationOptions animationOptions(mbgl::Milliseconds(duration * 1000));
+        animationOptions.minZoom = util::min(startZoom, endZoom);
+        animationOptions.transitionFinishFn = [&]() { transitionFinished = true; };
+
+        map.flyTo(mbgl::CameraOptions().withCenter(endPos).withZoom(endZoom), animationOptions);
+
+        for (; !transitionFinished; frames++) {
```

Comment:
nit: for (size_t frames = 0; !transitionFinished; ++frames) {

Reply:
`frames` variable is used outside of the for-loop, hence the wider scope.